src/components/PeopleFilters.tsx

```diff
@@ -1,18 +1,59 @@
-export const PeopleFilters = () => {
+import { useCallback, useEffect, useState } from 'react';
+import { SearchLink } from './SearchLink';
+import classNames from 'classnames';
+import { useSearchParams } from 'react-router-dom';
+
+export const PeopleFilters: React.FC = () => {
+  const [nameFilter, setNameFilter] = useState<string>('');
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const sex = searchParams.get('sex');
+  const centuries = searchParams.getAll('centuries');
+
+  // If the query is empty, remove it from the URL
+  useEffect(() => {
+    if (!nameFilter.length) {
+      searchParams.delete('query');
+      setSearchParams(searchParams);
+    } else {
+      searchParams.set('query', nameFilter);
+      setSearchParams(searchParams);
+    }
+  }, [nameFilter, searchParams, setSearchParams]);
+
+  const getCenturies = useCallback(
+    (century: string) => {
+      // If the century is selected now
+      return centuries.includes(century)
+        ? centuries.filter((currentCent: string) => currentCent !== century)
+        : [...centuries, century];
+    },
+    [centuries],
+  );
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">
+        <SearchLink
+          className={classNames({ 'is-active': sex === null })}
+          params={{ sex: null }}
+        >
           All
-        </a>
-        <a className="" href="#/people?sex=m">
+        </SearchLink>
+        <SearchLink
+          className={classNames({ 'is-active': sex === 'm' })}
+          params={{ sex: 'm' }}
+        >
           Male
-        </a>
-        <a className="" href="#/people?sex=f">
+        </SearchLink>
+        <SearchLink
+          className={classNames({ 'is-active': sex === 'f' })}
+          params={{ sex: 'f' }}
```

Comment:
Since you are working with TS, will be better to create a simple enum for genders to avoid somehow 'magical' null | 'f' | 'm' values.